packages/autocomplete-core/src/types/plugins.ts

```diff
@@ -0,0 +1,17 @@
+import { AutocompleteOptions, AutocompleteSource } from './api';
+
+export type AutocompletePlugin<TItem, TData = unknown> = Partial<
+  Pick<AutocompleteOptions<TItem>, 'onStateChange' | 'onSubmit' | 'getSources'>
+> & {
+  /**
+   * The subscribed properties are properties that are called when other sources
+   * are interacted with.
+   */
+  subscribed: {
+    onSelect: AutocompleteSource<TItem>['onSelect'];
```

Comment:
is there a reason the other events aren't here? onSubmit sounds the same as onSelect in my naive thought

Reply:
Just because I was mainly refactoring the Recent Searches plugin but yes—we'll support most of them here.